hack/generator/pkg/astmodel/testdata/SetOptionalObjectFromOptionalObject-Direct.golden

```diff
@@ -0,0 +1,48 @@
+// Code generated by k8s-infra. DO NOT EDIT.
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package vCurrent
+
+import "github.com/Azure/k8s-infra/hack/generated/_apis/Verification/vHub"
+
+type Person struct {
+	role *Release `json:"role"`
+}
+
+// ConvertFrom populates our Person from the provided source Person
+func (person Person) ConvertFrom(source *vHub.Person) error {
+
+	// role
+	if source.role != nil {
+		var role Release
+		err := role.ConvertFrom(source.role)
+		if err != nil {
+			return errors.Wrap(err, "populating role from role, calling ConvertFrom()")
+		}
+		person.role = &role
+	} else {
+		person.role = nil
```

Comment:
Ah, of course you need to explicitly assign nil here in case the destination wasn't empty.